lib/datadog/core/transport/http/config.rb

```diff
@@ -0,0 +1,258 @@
+# frozen_string_literal: true
+
+require 'json'
+require 'base64'
+
+require_relative '../config'
+require_relative 'client'
+require_relative '../../../../ddtrace/transport/http/response'
+require_relative '../../../../ddtrace/transport/http/api/endpoint'
+
+# TODO: Decouple standard transport/http/api/instance
+#
+# Separate classes are needed because transport/http/trace includes
+# Trace::API::Instance which closes over and uses a single spec, which is
+# negotiated as either /v3 or /v4 for the whole API at the spec level, but we
+# need an independent toplevel path at the endpoint level.
+#
+# Separate classes are needed because of `include Trace::API::Instance`.
+#
+# Below should be:
+# require_relative '../../../../ddtrace/transport/http/api/instance'
+require_relative 'api/instance'
+# Below should be:
+# require_relative '../../../../ddtrace/transport/http/api/spec'
+require_relative 'api/spec'
+
+module Datadog
+  module Core
+    module Transport
+      module HTTP
+        # HTTP transport behavior for remote configuration
+        module Config
+          # Response from HTTP transport for remote configuration
+          class Response
+            include Datadog::Transport::HTTP::Response
+            include Core::Transport::Config::Response
+
+            def initialize(http_response, options = {})
+              super(http_response)
+
+              # TODO: these fallbacks should be improved
+              roots = options[:roots] || []
+              targets = options[:targets] || ''
+              target_files = options[:target_files] || []
+              client_configs = options[:client_configs] || []
+
+              raise TypeError.new(Array, roots) unless roots.is_a?(Array)
+
+              @roots = roots.map do |e|
+                raise TypeError.new(String, e) unless e.is_a?(String)
+
+                begin
+                  decoded = Base64.strict_decode64(e) # TODO: unprocessed, don't symbolize_names
+                rescue ArgumentError
+                  raise DecodeError.new(:roots, e)
+                end
+
+                begin
+                  parsed = JSON.parse(decoded)
+                rescue JSON::ParserError
+                  raise ParseError.new(:roots, e)
+                end
```

Comment:
Do we want to capture the specific error here, or do we want to provide the `root` instance?

Reply:
the root value (see the `ParseError#initialize` arguments)